packages/localization/src/dates/modifyDateBy.ts

```diff
@@ -39,4 +39,59 @@ const modifyDateBy = (date: CalendarDate, amount: number, unit: string, minDate?
 	return newDate;
 };
 
+/**
+ * Adds or subtracts a given amount of days/months/years from a date.
+ * If minDate or maxDate are given, the result will be enforced within these limits
+ *
+ * <b>Note:</b> Used in the case when user is navigating with the mouse through the arrows of the Calendar header.
+ *
+ * @param date CalendarDate instance
+ * @param amount how many days/months/years to add (can be a negative number)
+ * @param unit what to modify: "day", "month" or "year"
+ * @param minDate minimum date to enforce
+ * @param maxDate maximum date to enforce
+ */
+const modifyDateByClick = (date: CalendarDate, amount: number, unit: string, minDate?: CalendarDate, maxDate?: CalendarDate) => {
```

Comment:
They are not quite the same. The `modifyDateBy` method's logic works ( after applying the code in `DayPicker.ts` in between lines 378 and 382 ) the way we want it to work by using specifically PageUp / PageDown buttons. In other words **the focus stays at the same day number, but only the month is changed ( f.e 14 May -> 14 June )** Which is the expected behavior for PgUp && PgDown.
On the other hand, the expected behavior by navigating through the Calendar's header by using the the arrow buttons and clicking on them with the mouse is the following - When the user clicks on the **"Next"** button, **the focus should apply to the first day of the next month**, and when the user clicks on the **"Previous"** button **the focus should apply to the last day of the previous month**. And since the `modifyDateBy` method works perfectly fine in case of using PageUp and PageDown I created separated method with slightly different logic for the cases where users change months with the header.

In terms of documentation in the `modifyDateByClick` method we have `* <b>Note:</b> Used in the case when user is navigating with the mouse through the arrows of the Calendar header.` , but I believe I should make the documentation more visible perhaps. Thank you for your comment.

Reply:
Hi @hinzzx,

Yes, I get it now. It seems to me though that we need a different kind of function for the arrow buttons scenario as only the day view is affected. For the year and month views the focus would stay at the same grid cell (same as the Page Up/Down interactions). The _showNextPage/_showPreviousPage functions are called when a user pressed over the arrow buttons in the calendar header. In their documentation it is assumed that the behavior would be the same as with Page Up/Down, which turns out to not be correct.
The `modifyDateBy` function documentation is clear as it only modifies by a given unit. The `modifyDateByClick` however does something very specific with the date modification in the month case, which is not clear by the documentation. Also there is no change for the day and year case.
Theoretically we could use `modifyDateBy` twice for the arrow button press scenario:
- once to change the month to the next or previous month
- once to change the day to the first or last day of the month (In order to do that we could use `getDaysInMonth` method

@ilhan007 could we also have your opinion for that.